### ISSUE TITLE
Random segfaults / access violations on Windows

### ISSUE THREAD

**Issue by jaimergp** [mamba-org/mamba#3642]
### Troubleshooting docs

- [X] My problem is not solved in the Troubleshooting docs

### Anaconda default channels

- [X] I do NOT use the Anaconda default channels (pkgs/* etc.)

### How did you install Mamba?

Mambaforge or latest Miniforge

### Search tried in issue tracker

windows segfaults

### Latest version of Mamba

- [X] My problem is not solved with the latest version

### Tried in Conda?

I have this problem with Conda as well, without using Mamba

### Describe your issue

The conda and conda-libmamba-solver CI has seen random segfaults on the Windows CI for a while. See https://github.com/conda/conda-libmamba-solver/issues/334 for some examples.

They have been happening for a while in 1.x, but seldomly. Now they are more frequent with 2.x. I compiled a few from the recent weeks. It's not super consistent and somehow random, so It Will Be Tricky (tm).

Things I've noticed:

- They are not tied to any particular test
- They are not tied to any particular Python version
- They are not tied to any particular libmamba version (I've seen it in several versions across the 1.x and 2.x series)

### mamba info / micromamba info

```shell
N/A
```

### Logs

```shell
See https://github.com/conda/conda-libmamba-solver/issues/334
```

### environment.yml

```yaml
N/A
```

### ~/.condarc

```yaml
N/A
```

**Comment by Klaim** [comment]
@jaimergp If we provide you a custom release-with-debug-infoversion of the libmamba(py) binaries, would you be able to try it in your ci? So far I'm failing to reproduce these random crashes without dubious builds setup and I failed to reproduce it with conda-libmamba-solver tests but I suspect I'm doing something different (and some tests fails locally for completely unrelated reasons). We need more info and I'm looking for ways to get these info with a more consistent repo-case.
We do have some other random failures on our CI but they dont seem related to segfaults.

**Comment by Klaim** [comment]
Note: I tried with ASAN builds but unfortunately it's full of potential false-positive unless all the dependencies (coming from conda packages) are also built with ASAN (except crt). I'm trying to find another way to achieve that meanwhile, probably using vcpkg but it's a bit buggy at the moment.

**Comment by jaimergp** [comment]
@Klaim Yep, I can instrument our CI to install from a custom channel/label or something. Let me know how we can help!

**Comment by Klaim** [comment]
@jaimergp I'm preparing a special build for you 👍🏽, let me know how I can send you a link privately to the archive (I didnt find your email).
